clippy_lints/src/methods/mod.rs

```diff
@@ -3563,6 +3564,43 @@ declare_clippy_lint! {
     "calls to `.take()` or `.skip()` that are out of bounds"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for calls to `Path::join` that start with a path separator, like `\\` or `/`..
+    ///
+    /// ### Why is this bad?
+    /// `.join()` arguments starting with a separator  (`/` or `\\`) can replace the entire path.
+    /// If this is intentional, prefer using `Path::new()` instead.
```

Comment:
```suggestion
    /// If the argument to `Path::join` starts with a separator, this will overwrite the original path. If this is intentional, prefer using `Path::new` instead.
```